### PR TITLE
upgrade Istio add-on to v1.6.4

### PR DESCRIPTION
**Pull request by tegk**
Istio v1.6.0 is a major update that should be available to a broad audience via microk8s.

### CODE REVIEW

**Comment by balchua** [comment]
Thanks for this PR @tegk.

 I do have some thoughts about using `istioctl` to install istio, just like what istio documentation says.  The same thing with disabling the addon.
Instead of applying the `demo` profile we can have users set profiles while enabling the addon.
For example:
`microk8s enable istio "profile=default:addonComponents.grafana.enabled=true"`

Where these `profile=default:addonComponents.grafana.enabled=true` are then parsed to become

`istioctl install --set profile=default --set addonComponents.grafana.enabled=true`

Just my 2 cents.

What do you think?

**Comment by tegk** [comment]
> Thanks for this PR @tegk.
> 
> I do have some thoughts about using `istioctl` to install istio, just like what istio documentation says. The same thing with disabling the addon.
> Instead of applying the `demo` profile we can have users set profiles while enabling the addon.
> For example:
> `microk8s enable istio:profile=default;addonComponents.grafana.enabled=true`
> 
> Where these `profile=default;addonComponents.grafana.enabled=true` are then parsed to become
> 
> `istioctl install --set profile=default --set addonComponents.grafana.enabled=true`
> 
> Just my 2 cents.
> 
> What do you think?

@balchua Sounds like a good idea! Could you open an issue where we can discuss this?

**Comment by balchua** [comment]
We can continue the discussion in the istio upgrade issue https://github.com/ubuntu/microk8s/issues/1222

**Comment by balchua** [comment]
Some few pointers.

When transitioning from the old yamls to the new, we may have to make sure that we disable the old resources together with the new resources created.

Otherwise the old resources will stay there.

**Comment by tringuyen-yw** [comment]
OK to update to [Istio 1.6.3](https://github.com/istio/istio/releases) released on 2020-06-18 ?

**Comment by ktsakalozos** [comment]
@tringuyen-yw, sure. The latest the better.

@balchua @tegk, I guess you do not want me to wait for the istioctl based installation, right?

**Comment by balchua** [comment]
@ktsakalozos im fine with the istio 1.6 PR to push through.  We can work on the `istioctl` based install for later versions.
@tegk what do you think?

**Comment by tegk** [comment]
@tringuyen-yw,I have updated the version to 1.6.3
@ktsakalozos Please merge in case you do not see any problems as this would hugely beneficia

**Comment by tringuyen-yw** [comment]
Thanks very much for taking care of the issue. Just curious, is it that simple to upgrade an add-on version?

**Comment by tegk** [comment]
> Thanks very much for taking care of the issue. Just curious, is it that simple to upgrade an add-on version?

Yes, its trivial. Just change the version number in the files that were changed in this PR.

File: snap/hooks/configure [hunk 1]
```diff
@@ -159,7 +159,7 @@ fi
 
 if ([ -f "$SNAP_USER_COMMON/istio-auth.lock" ] || [ -f "$SNAP_USER_COMMON/istio-auth.lock" ]) && ! [ -f "$SNAP_DATA/bin/istioctl" ]
 then
-  ISTIO_VERSION="v1.0.5"
```
Comment (ktsakalozos) [inline snap/hooks/configure:162]: Leave this to 1.0.5 please.

**Comment by sstubbs** [comment]
Hi,

I'm not sure this information will help anyone but just adding it in case.

I've just installed istio manually as I want a minimal installation. The demo profile uses a lot of resources and I would rather not have two copies of prometheus running.

Here is what I've installed.

```
istioctl install \
--set profile=default \
--set values.grafana.enabled=false \
--set values.prometheus.enabled=false \
--set values.gateways.istio-egressgateway.autoscaleEnabled=false \
--set values.gateways.istio-ingressgateway.autoscaleEnabled=false \
--set values.pilot.autoscaleEnabled=false \
--set meshConfig.enablePrometheusMerge=true
```

And then I manually add the grafana charts to prometheus. When I asked the istio commutnity a few months ago I was told the integrated prometheus is not for production purposes and they now provide the charts for other grafana installs https://grafana.com/grafana/dashboards?search=istio.

**Comment by tringuyen-yw** [comment]
@sstubbs 

Q1. can you confirm that by "installed istio manually" you meant installing istio from binaries, outside of microk8s, using the procedure described in https://istio.io/latest/docs/setup/getting-started/ ?

Q2. After your manual install of istio, what is the output of `microk8s status | grep istio`?

**Comment by sstubbs** [comment]
Yes that is how I installed it.

Here are all the addons I enabled before hand for reference.
```
sudo snap install microk8s --classic --channel=latest/edge/ha-preview
microk8s enable ha-cluster dns rbac prometheus
```

When I run `microk8s status | grep istio` I get `istio: enabled`

**Comment by tringuyen-yw** [comment]
`istio: enabled` even though you didn't explicitly enable the add-on via `microk8s enable istio` ???

I don't know how microk8s manages the "foreign" istio and could somehow recognize as its own add-on. I would like the opinion of microk8s experts on this. Thanks

**Comment by sstubbs** [comment]
Yes it is strange. I did have another mirok8s snap installed previously but then I ran.
```
microk8s reset
sudo snap remove microk8s
```
Then I installed this. Could it be cached somewhere from that previous installation.

**Comment by balchua** [comment]
Hi microk8s detect addons present is mostly be based on if a cluster resource is present or if the binary exist in certain locations, for example `${SNAP_DATA}/bin/helm`.
In the case of istio it checks for the pod name `istio`.
The list of addons are kept in this file `https://github.com/ubuntu/microk8s/blob/master/microk8s-resources/wrappers/addon-lists.yaml`.

**Comment by tegk** [comment]
Istio 1.6.4 is out with security fixes that also affects the version that is currently used in Microk8s

**Comment by tegk** [comment]
We need to use the new istioctl installing process so I am closing this until I have time to actually contribute this.

**Pull request closed without merging**